### ISSUE TITLE
Check unused variables in the global namespace

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#919]
If we have unused variables at the module level, we don't warn about them, since we're
considering them to be part of an API. For instance, something like this won't emit unused-variable:

``` python
base, ext = os.path.splitext(...)
if ext not in ...:
    pass
```

While this is good by default, in some cases we need more control. We should add a new flag, something of `--global-unused` or sorts, that could check unused variables at the global level.

**Comment by PCManticore** [comment]
Or maybe ``--allow-global-unused-variables`` set by default to true.
